types/relay-compiler/lib/core/GraphQLIRVisitor.d.ts

```diff
@@ -109,4 +109,29 @@ export type NodeVisitor =
           Variable?: NodeVisitorObject<Variable>;
       };
 
-export function visit(root: VisitNode, visitor: NodeVisitor): any;
+export type EnumTypeID = any;
+export type FieldID = any;
+export type ScalarTypeID = any;
+export type TypeID = any;
+
+export interface Schema {
+    expectStringType: () => ScalarTypeID;
+    getEnumValues: (type: EnumTypeID) => string[];
+    getFieldName: (field: FieldID) => string;
+    getFields: (type: TypeID) => FieldID[];
+    getFieldType: (field: FieldID) => TypeID;
+    getListItemType: (type: TypeID) => TypeID;
+    getNullableType: (type: TypeID) => TypeID;
+    getTypeString: (type: TypeID) => string;
+    isAbstractType: (type: TypeID) => boolean;
+    isEnum: (type: TypeID) => boolean;
+    isInputObject: (type: TypeID) => boolean;
+    isInterface: (type: TypeID) => boolean;
+    isList: (type: TypeID) => boolean;
+    isNonNull: (type: TypeID) => boolean;
+    isObject: (type: TypeID) => boolean;
+    isScalar: (type: TypeID) => boolean;
+    isUnion: (type: TypeID) => boolean;
```

Comment:
What about making all these predicate methods TS type guards, as per @smmoosavi’s comment [here](https://github.com/relay-tools/relay-compiler-language-typescript/pull/154#discussion_r339107825)?

Reply:
@alloy I think it's a great suggestion, and something we should definitely do, however it will work without it for now, and this PR really is blocking me from completing the PR in relay-compiler-language-typescript to add support for v7. So if possible, could we make a separate PR for type guards and their types/interfaces? Would really be nice to have this one merged as soon as possible, so we can continue the work on https://github.com/relay-tools/relay-compiler-language-typescript/pull/154 - I'd like to look into this type guard thing afterwards, if nobody else is eager to do so.

Reply:
Sounds good 👍